dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/message/LengthFieldStreamingDecoder.java

```diff
@@ -130,16 +127,12 @@ private void deliver() {
     }
 
     private void processHeader() throws IOException {
-        ByteArrayOutputStream bos = new ByteArrayOutputStream(lengthFieldOffset + lengthFieldLength);
         byte[] offsetData = new byte[lengthFieldOffset];
         int ignore = accumulate.read(offsetData);
-        bos.write(offsetData);
         processOffset(new ByteArrayInputStream(offsetData), lengthFieldOffset);
         byte[] lengthBytes = new byte[lengthFieldLength];
         ignore = accumulate.read(lengthBytes);
-        bos.write(lengthBytes);
         requiredLength = bytesToInt(lengthBytes);
-        this.dataHeader = new ByteArrayInputStream(bos.toByteArray());
 
         // Continue reading the frame body.
         state = DecodeState.PAYLOAD;
```

Comment:
**Performance Issue**: The method processHeader inefficiently processes header data by reading into byte arrays without considering the impact on memory and CPU usage, especially in high-throughput scenarios. <br> **Fix**: Optimize the processHeader method to minimize memory allocations and improve CPU efficiency. Consider using a buffer pool to reuse byte arrays and reduce garbage collection pressure. <br> **Code Suggestion**: 
 ```
 -        byte[] offsetData = new byte[lengthFieldOffset];
 -        int ignore = accumulate.read(offsetData);
 -        processOffset(new ByteArrayInputStream(offsetData), lengthFieldOffset);
 -        byte[] lengthBytes = new byte[lengthFieldLength];
 -        ignore = accumulate.read(lengthBytes);
 -        requiredLength = bytesToInt(lengthBytes);
 +        ByteBuffer buffer = ByteBuffer.allocate(lengthFieldOffset + lengthFieldLength);
 +        accumulate.read(buffer.array(), 0, lengthFieldOffset + lengthFieldLength);
 +        processOffset(new ByteArrayInputStream(buffer.array(), 0, lengthFieldOffset), lengthFieldOffset);
 +        requiredLength = bytesToInt(Arrays.copyOfRange(buffer.array(), lengthFieldOffset, lengthFieldOffset + lengthFieldLength));
 ```